core/arch/arm/plat-stm32mp1/nsec-service/bsec_svc.c

```diff
@@ -32,11 +32,17 @@ void bsec_main(struct thread_smc_args *args)
 		break;
 	case STM32_SIP_SVC_BSEC_PROG_OTP:
 		FMSG("program @%#"PRIx32, otp_id);
-		result = stm32_bsec_program_otp(in_value, otp_id);
+		if (IS_ENABLED(CFG_STM32_BSEC_WRITE))
+			result = stm32_bsec_program_otp(in_value, otp_id);
+		else
+			DMSG("OTP programming not supported");
 		break;
 	case STM32_SIP_SVC_BSEC_WRITE_SHADOW:
 		FMSG("write shadow @%#"PRIx32, otp_id);
-		result = stm32_bsec_write_otp(in_value, otp_id);
+		if (IS_ENABLED(CFG_STM32_BSEC_WRITE))
+			result = stm32_bsec_write_otp(in_value, otp_id);
+		else
+			DMSG("OTP programming not supported");
```

Comment:
Wouldn't it be easier to keep the old inline function and let that print this?

Reply:
Ok, I think this last commit is useless.
I'll discard it.
Thanks for the feedback.